test/RTreeProperties.hs

```diff
@@ -1,145 +1,70 @@
+import Data.Function               (on)
+import Data.List                   ((\\))
+import Data.RTree.Base
+import Data.RTree.MBB              hiding (mbb)
+import HaskellWorks.Hspec.Hedgehog
+import Hedgehog
+import Prelude                     hiding (length, lookup, map, null)
+import Test.Hspec
+import Test.HUnit                  hiding (Test, Testable)
+import Text.Show.Functions         ()
 
+import qualified Data.List as L (length)
+import qualified Gen       as G
 
 --import Graphics.Gnuplot.Simple
 
 -- todo: write tests
 
 main :: IO ()
-main = do
-    defaultMain
-       [
-             testCase "test_null" test_null
-           , testCase "test_singleton" test_singleton
-           , testCase "test_insert" test_insert
-           , testCase "test_lookup" test_lookup
-           , testCase "test_lookupRange" test_lookupRange
-           , testCase "test_lookupRangeWithKey" test_lookupRangeWithKey
-           , testCase "test_lookupContainsRange" test_lookupContainsRange
-           , testCase "test_lookupContainsRangeWithKey" test_lookupContainsRangeWithKey
-           , testCase "test_union" test_union
-           , testCase "test_unionWith" test_unionWith
-           , testCase "test_length" test_length
-           , testCase "test_keys" test_keys
-           , testCase "test_values" test_values
-           , testCase "test_delete" test_delete
-           , testCase "test_fromList" test_fromList
-           , testCase "test_binary" test_binary
-           , testProperty "prop_mbb" prop_mbb
-           , testProperty "prop_rtree" prop_rtree
-           
---       , testProperty "map a StringMap" prop_map
-
-       ]
--- ------------------------
-t_mbb1, t_mbb2 , t_mbb3, t_mbb4, t_mbb5, t_mbb6, t_mbb7, t_mbb8 :: MBB
-t_mbb1 = (MBB 0.0 0.0 1.0 1.0)
-t_mbb2 = (MBB 5.0 0.0 6.0 1.0)
-t_mbb3 = (MBB 1.0 2.0 2.0 3.0)
-t_mbb4 = (MBB 6.0 2.0 7.0 3.0)
-t_mbb5 = (MBB 3.0 3.0 4.0 4.0)
-t_mbb6 = (MBB 0.0 0.0 0.0 0.0)
-t_mbb7 = (MBB 1.0 2.0 5.0 4.0)
-t_mbb8 = (MBB 4.0 0.0 6.0 3.0)
-
-t_1, t_2, t_3 :: RTree String
-t_1 = singleton t_mbb1 "a"
-t_2 = singleton t_mbb2 "b"
-t_3 = singleton t_mbb3 "c"
-
-
-u_1, u_2, u_3 :: [(MBB, String)]
-u_1 = [(t_mbb1, "a"), (t_mbb2, "b"),(t_mbb3, "c"),(t_mbb4, "d")]
-u_2 = [(t_mbb5, "e"), (t_mbb6, "f")] ++ u_1
-u_3 = [(t_mbb7, "g"), (t_mbb8, "h")] ++ u_2
-
-tu_1, tu_2, tu_3 :: RTree String
-tu_1 = fromList u_1
-tu_2 = fromList u_2
-tu_3 = fromList u_3
-
--- ------------------------
-eqRt :: (Show a, Eq a) => RTree a -> RTree a -> Assertion
-eqRt = eqList `on` toList
-
-eqList :: (Show a, Eq a) => [a] -> [a] -> Assertion
-eqList l1 l2 = do
-    [] @=? (l1 \\ l2)
-    (L.length l1) @=? (L.length l2)
--- ------------------------
-
-test_null :: Assertion
-test_null = do
-    null empty @?= True
-    null t_1 @?= False
-
-test_singleton :: Assertion
-test_singleton = do
+main = hspec $ do
+  it "test_null" $ do
+    null empty  @?= True
+    null t_1    @?= False
+  it "test_singleton" $ do
```

Comment:
Please give me some time to have a look at hspec. At first, it looks to be a bit strange to have all the tests in one big main function. I need to get used to it!

Reply:
It's possible to have them separate as well if you prefer or if you want to parameterise the test, although the support for working with tests when structured in the single function manner is very good.

For example you can change `it` to `fit` to make it the only test to run so you can concentrate on fixing a particular failing test.  Or change `it` to `xit` to skip running that test.